### PR TITLE
Flip min/max if necessary

### PR DESCRIPTION
**Pull request by heltluke**

### CODE REVIEW

File: pyrpl/software_modules/lockbox/models/fabryperot.py [hunk 1]
```diff
@@ -96,9 +96,12 @@ def fit(self, curve, voltage):
...
                          max_nfev=self.max_nfev)
-
-        maximum = result.values["height"] + result.values["c"]
-        minimum = result.values["c"]
+        if not negative:
+            maximum = result.values["height"] + result.values["c"]
+            minimum = result.values["c"]
+        else:
+            maximum = result.values["c"]
+            minimum = result.values["c"] + result.values["height"]
```
Comment (lneuhaus) [inline pyrpl/software_modules/lockbox/models/fabryperot.py]: first time I get to use the XOR operator in python - maybe too much of a stretch?
```suggestion
        minimum = result.values["c"]
        maximum = minimum + result.values["height"]
        if self.negative:
            minimum, maximum = maximum, minimum
        if minimum > maximum:
            raise FitError("The fitted peak has the wrong sign.")
```

File: pyrpl/software_modules/lockbox/models/fabryperot.py [hunk 2]
```diff
@@ -96,9 +97,12 @@ def fit(self, curve, voltage):
...
                          max_nfev=self.max_nfev)
-
-        maximum = result.values["height"] + result.values["c"]
-        minimum = result.values["c"]
+        if not self.negative:
+            maximum = result.values["height"] + result.values["c"]
+            minimum = result.values["c"]
+        else:
+            maximum = result.values["c"]
+            minimum = result.values["c"] + result.values["height"]
```
Comment (lneuhaus) [inline pyrpl/software_modules/lockbox/models/fabryperot.py:105]:
```suggestion
        minimum = result.values["c"]
        maximum = minimum + result.values["height"]
        if self.negative:
            minimum, maximum = maximum, minimum
        if minimum > maximum:
            raise FitError("The fitted peak has the wrong sign.")
```

File: pyrpl/software_modules/lockbox/models/fabryperot.py [hunk 3]
```diff
@@ -45,6 +45,10 @@ class FitInput(InputSignal):
     
     def calibrate(self, autosave=False):
         curve, voltage = self.sweep_acquire(return_asg_data=True)
+        points_to_use = len(curve)//2
+        curve = curve[:points_to_use]
+        voltage = voltage[:points_to_use]
+        curve -= self.calibration_data._analog_offset
```
Comment (lneuhaus) [inline pyrpl/software_modules/lockbox/models/fabryperot.py:51]:
```suggestion
```
Already subtracted by sweep_acquire, and added back to compute the setpoint.

File: pyrpl/software_modules/lockbox/models/fabryperot.py [hunk 4]
```diff
@@ -45,6 +45,9 @@ class FitInput(InputSignal):
     
     def calibrate(self, autosave=False):
         curve, voltage = self.sweep_acquire(return_asg_data=True)
+        points_to_use = len(curve)//2
```
Comment (heltluke) [inline pyrpl/software_modules/lockbox/models/fabryperot.py:48]: :+1:

**Pull request merged**